### ISSUE TITLE
Is this package still being maintained?

### ISSUE THREAD

**Issue by Javiani** [pugjs/pug-loader#124]
Hi!

I see that there aren't many stars but there are a lot of people using this package, but it has being a while since it was updated for the last time...

I'm using pug today and it's very useful in many cases, and I'm trying to use it with storybook but there's a flaw on this loader... I can't import a file that has a mixin on it. It always returns a empty template.

Man, that bug it's killing me... Please don't let this project die... It fells like there's no other option than those complicated framework out there... Pug is way better and simpler... please do not give up on this package... there are too many people using it...

**Comment by matheo** [comment]
@TimothyGu seems active but working on other projects :(

**Comment by alexander-akait** [comment]
/cc @ForbesLindesay We want to provide official supporting from webpack team to pug and move this repo to `webpack-contrib`? Can you help us?

We need:

1. You need to send a PR with updating license on JS Foundation (https://github.com/webpack-contrib/css-loader/blob/master/LICENSE), we are not trying to steal the code, we just want to protect our users from the fact that in the future the original developer may want to change the license, you can save your name in any other places (package.json/comments in code/other, we never remove it)
2. Give me github and npm permissions
3. I will more this repo to `webpack-contrib`
4. We resolve issues and do new major release.

**Comment by matheo** [comment]
awesome stuff @alexander-akait 
I hope you get an answer :)

**Comment by alexander-akait** [comment]
Friendly ping @ForbesLindesay and @TimothyGu

**Comment by spence-s** [comment]
This would be great and give a little more piece of mind to everyone depending on this loader. Hope it gets migrated soon!

**Comment by alexander-akait** [comment]
No answer from developers :disappointed:

**Comment by alexander-akait** [comment]
I could have done it long ago

**Comment by Rush** [comment]
Agreed this would be great.  @alexander-akait - could the original developers be tracked elsewhere? For example LinkedIn or email (extracted from commits).

**Comment by alexander-akait** [comment]
@Rush I tried to connect using twitter, but no luck

**Comment by mrmowgli** [comment]
The current license suggests that moving it would be fine.  Also it looks like it was originally maintained by someone else. The only downside to moving it is no-one will update this package and list it as unmaintained or to look for it at webpack-contrib.

**Comment by alexander-akait** [comment]
Written @ForbesLindesay in twitter DM again, maybe we will succeed

**Comment by Rush** [comment]
He actually appears to be active on Github in general. I also sent him an email and linked this thread.

**Comment by m10** [comment]
maybe @sokra can help? 
he startet the project in 2012 https://github.com/pugjs/pug-loader/commit/02e8aa9d91409c2d5fc9cffd7e12b78615cb2bdd

**Comment by alexander-akait** [comment]
@alubbe @bloodyowl @christopherdowson @droooney @ezhlobo @ForbesLindesay @hemanth @jbnicolai @leenty @LionNatsu  @rlidwka @TimothyGu @tjconcept Is anyone here? Sorry for spam, I can not contact anyone

**Comment by christopherdowson** [comment]
@alexander-akait Sorry, I only created the logo. I've not worked with anyone on the project in a few years.

**Comment by sokra** [comment]
> maybe @sokra can help?

I have npm permission, but I don't have github permission. If someone could give @alexander-akait or me admin access to this repo we could move it over or keep it here and allow someone to maintain it here.

**Comment by m10** [comment]
Any updates on this?
It's been 6 months since @alexander-akait requesting maintenance https://github.com/pugjs/pug-loader/issues/124#issuecomment-685641183 .
No activity from maintainers, no commits since 3+ years etc. The chances of getting hold of them seem slim?
What are the options to go forward without the help of the maintainers?
Could you fork it? 😱
I'd love to use pug with webpack, but without a proper loader that is being maintained this is not feasible with the outstanding issues at the moment 😞

**Comment by matheo** [comment]
And recently I had issues with `pug-lint` outdated dependencies :scream:

**Comment by tjconcept** [comment]
I've added @sokra as an admin. I have had next to nothing to do with Pug and it's been years since I've been writing any serious amounts of HTML, although I still occasionally update a project using Pug.

**Comment by sokra** [comment]
Thanks

**Comment by spence-s** [comment]
In the meantime - I forked and published [simple-pug-loader](https://www.npmjs.com/package/simple-pug-loader) - if anyone was having similar problems as me with this loader.

**Comment by alexander-akait** [comment]
Ping again @alubbe @bloodyowl @christopherdowson @droooney @ezhlobo @ForbesLindesay @hemanth @jbnicolai @leenty @LionNatsu @rlidwka @TimothyGu

**Comment by christopherdowson** [comment]
Please don't ping me again if possible. I just did the logo in, like, 2016.

**Comment by bloodyowl** [comment]
Likewise, not in this org anymore.

**Comment by noahbrenner** [comment]
@alexander-akait From the recent comments, it looks like @sokra was given repo permission a week or so ago, so he might be the one to coordinate with.

**Comment by sokra** [comment]
I only have write access. I can't move or give anyone else access.

**Comment by noahbrenner** [comment]
@sokra Ah, that makes sense. In case it's helpful for tracking someone down to make the change, here's a way you can look up who's an admin: https://stackoverflow.com/questions/42538330/find-out-who-is-the-admins-for-a-git-repo/42539102

I did this with another repo earlier today, and it worked to just pass the endpoint to `$ gh api`. That way, you don't have to manually pass an access token.

**Comment by sokra** [comment]
Ok here are the admins:

```
          {
              "login": "hemanth",
              "name": "hemanth.hm"
          },
          {
              "login": "tjconcept",
              "name": "Thomas Jensen"
          },
          {
              "login": "ForbesLindesay",
              "name": "Forbes Lindesay"
          },
          {
              "login": "TimothyGu",
              "name": "Timothy Gu"
          },
          {
              "login": "jbnicolai",
              "name": "Joshua Appelman"
          },
 ```

**Comment by noahbrenner** [comment]
Hi, `pug-loader` repo admins! @hemanth @tjconcept @ForbesLindesay @TimothyGu @jbnicolai

What do you think of [alexander-akait's](https://github.com/pugjs/pug-loader/issues/124#issuecomment-685641183) and [sokra's](https://github.com/pugjs/pug-loader/issues/124#issuecomment-765373940) offer of the webpack team supporting `pug-loader` and moving it to `webpack-contrib`?

There have been some issues unaddressed for a while, particularly changing the version range of the `pug` dependency to support the newest version. It sounds like the webpack folks would love to take that maintenance work off your hands.

---

@tjconcept:

> I've added @sokra as an admin.

It's looking like you've added him with write privileges, but not as an admin. Would you be up for giving him admin access? Does there need to be more discussion between current admins first?

**Comment by sokra** [comment]
Note that even with admin access I will only move this repo if there is an official "go" from the maintainers.

**Comment by tjconcept** [comment]
I won't move it as I never got such mandate from the author.

I think the suggestion for the webpack team to maintain it is great, if it's valuable to them. I see no reason they can't go ahead with that.

**Comment by ForbesLindesay** [comment]
I'm happy to move this package to webpack-contrib. I don't (unfortunately) have permission to create a repo in that org, so cannot do it right now. I'm also happy to add someone from the webpack team to the npm package.

Please e-mail me (forbes@lindesay.co.uk) when you've given me permission to create repos in webpack-contrib. I'm also happy for any of the other maintainers to go ahead and move it if that's easier.

P.S. sorry for the delayed response. I've been very focused on other projects this year, and had un-watched this repo as I don't personally use pug & webpack together on any projects and there were several other maintainers listed.

**Comment by alexander-akait** [comment]
@ForbesLindesay Done https://github.com/webpack-contrib/pug-loader and assigned you to admin

**Comment by oliversalzburg** [comment]
I see the new repo is still not available yet. I wanted to open some issues and was wondering what the right place would be to work on this project.

**Comment by alexander-akait** [comment]
@ForbesLindesay Friendly ping

**Comment by m10** [comment]
> Please e-mail me (forbes@lindesay.co.uk) when you've given me permission to create repos in webpack-contrib. 

did someone send him an e-mail? He probably won't receive / read notifications since he unwatched the repo etc?

**Comment by noahbrenner** [comment]
@alexander-akait Did you try emailing him? As m10 pointed out, it sounds like he doesn't have notifications enabled for this repo and instead asked you to email after setting up the necessary permissions.

**Comment by alexander-akait** [comment]
@noahbrenner yes, multiple times

**Comment by Rush** [comment]
In case anybody is looking for a good alternative, I started using https://github.com/webdiscus/pug-loader (I am not affiliated!) which has great Webpack 5 support and reloading on every dependant file. And it even has multiple rendering modes. `html` mode is very useful for Angular. I just migrated my server-build templates and Angular app templates to this new loader and everything went fast and smooth.

**Comment by m10** [comment]
any update on this? is there some other way you can proceed with this? @ForbesLindesay seems unavailable..?
has someone else the necessary rights to move this repo (now that there is an "ok" from the author)?
Is it a possible alternative to use https://github.com/webdiscus/pug-loader as a basis instead?
@tjconcept @sokra @alexander-akait

**Comment by alexander-akait** [comment]
Looks like https://github.com/webdiscus/pug-loader is good solution

**Comment by Rush** [comment]
> Looks like https://github.com/webdiscus/pug-loader is good solution

Correct. The latest version of webdiscus/pug-loader 1.7.1 even works well with webpack asset pipeline. I can `require` icon assets directly from PUG and it will resolve proper locations.

**Comment by webdiscus** [comment]
Hello,

I have released new version v2.2.0 of the [pug-plugin](https://github.com/webdiscus/pug-plugin). This plugin already contains the [@webdiscus/pug-loader](https://github.com/webdiscus/pug-loader).

Why the pug-plugin?
- you can define pug template as webpack entry-point
```js
const PugPlugin = require('pug-plugin');
module.exports = {
  entry: {
    'index': './src/template/index.pug',
    'about': './src/template/pages/about/index.pug',
    // ...
  },
  plugins: [
    new PugPlugin(),
  ],
  module: {
    rules: [
      {
        test: /\.pug$/,
        loader: PugPlugin.loader,
      },
    ],
  },
  // ...
};
```
- source files of scss, js can be used directly in pug, do not need to define them in the webpack entry \
   all required resources in pug will be automatically processed by webpack
```pug
link(href=require('./styles.scss') rel='stylesheet')
script(src=require('./main.js'))
```
- generated HTML contains hashed CSS and JS output filenames
```html
<link href="/assets/css/styles.05e4dd86.css" rel="stylesheet">
<script src="/assets/js/main.f4b855d8.js"></script>
```
- resolves source files in SCSS/CSS and replace them with hashed output filenames
```scss
// resolve assets in imported package
@use 'material-icons'; 

// resolve webpack alias
@font-face {
  font-family: 'Montserrat';
  src: url('~Fonts/Montserrat/Montserrat-Regular.woff2'); //-> url(/assets/fonts/Montserrat-Regular.woff2)
}

// resolve relative path
.header {
  background-image: url('../images/header.png'); //-> url(/assets/img/header.4fe56ae8.png)
}
```

- pug-plugn can pretty format HTML
   the `pretty` option of original pug-loader is deprecated and not works

- test coverage 99% with many usage cases

You can see complete example [here](https://github.com/webdiscus/pug-plugin/tree/v2.2.0#usage-examples)
